srml/support/src/dispatch.rs

```diff
@@ -922,16 +924,16 @@ macro_rules! decl_module {
 					$fn_vis fn $fn_name (
 						$from $(, $param_name : $param )*
 					) $( -> $result )* {
-						// Check doughnut grants permission before dispatching the call
-						let doughnut_opt: Option<$crate::dispatch::Doughnut<$trait_instance::AccountId, $trait_instance::Signature>> = $crate::storage::unhashed::get(b":doughnut");
-
-						if let Some(doughnut) = doughnut_opt {
-							if doughnut.certificate.permissions.is_empty() {
-								return Err("Doughnut with empty permissions");
-							}
-							if doughnut.certificate.permissions[0].0.as_slice() != env!("CARGO_PKG_NAME").as_bytes() {
-								return Err("Doughnut does not grant permission for this domain");
-							}
+						use $crate::dispatch::DoughnutVerifier;
+						// Check if a doughnut exists in this execution context and whether it grants permission to
+						// dispatch the call.
+						if let Some(doughnut) = $crate::storage::unhashed::get(b"doughnut") {
+							// TO-DOUGH: Eventually we'll want to pass `$param_name` and `$param`
+							let _ = <T as system::Trait>::DoughnutVerifier::verify_doughnut(
+								&doughnut,
+								env!("CARGO_PKG_NAME"), // module
```

Comment:
in case you don't know, there is `module_path` macro. but I am not sure if it is better than `env!("CARGO_PKG_NAME")` for this case

Reply:
@xlc It seems in our case `env!` is the more direct solution